src/Scheduler.h

```diff
@@ -33,14 +41,11 @@ class CScheduler : public CInstrument
 
 	// Set that contains instruments to be currently processed
 	// Instruments will move in and out of this continuously
-	unordered_set<CInstrument*> m_SetInsts;
-
-	// All instrument pointers are placed and SHOULD STAY here to be deleted by the destructor
-	unordered_set<CInstrument*> m_GarbageCollector;
+	unordered_set<std::shared_ptr<CInstrument>> m_SetInsts;
 
 	// This can be viewed as the schedule's internal clock
-	int64_t m_iSampleCounter = 0;
-	int64_t m_iScheduleLength = 0;
+	std::atomic<int64_t> m_iSampleCounter = 0;
```

Comment:
Does this get modified from outside the audio thread?

Reply:
It can be used/read in a function called from the GUI thread. Definitely may not be necessary -- especially since we never call the function anyway lol. I'll leave it in for now